xmodule/tests/test_services.py

```diff
@@ -163,3 +166,84 @@ class TestTeamsConfigurationService(ConfigurationServiceBaseClass):
+
+    def setUp(self):
+        super().setUp()
+        self._xblock = mock.MagicMock()
+        now = datetime.now(UTC)
+        day_delta = timedelta(days=1)
+        self.yesterday = now - day_delta
+        self.today = now
+        self.tomorrow = now + day_delta
+
+    def test_show_correctness_default(self):
+        """
+        Test that correctness is visible by default.
+        """
+        assert ProblemFeedbackService(self._xblock).correctness_available()
+
+    @ddt.data(
+        (ShowCorrectness.ALWAYS, True),
+        (ShowCorrectness.ALWAYS, False),
+        # Any non-constant values behave like "always"
+        ('', True),
+        ('', False),
+        ('other-value', True),
+        ('other-value', False),
+    )
+    @ddt.unpack
+    def test_show_correctness_always(self, show_correctness, has_staff_access):
+        """
+        Test that correctness is visible when show_correctness is turned on.
+        """
+        self._xblock.show_correctness = show_correctness
+        assert ProblemFeedbackService(self._xblock, user_is_staff=has_staff_access).correctness_available()
+
+    @ddt.data(True, False)
+    def test_show_correctness_never(self, has_staff_access):
+        """
+        Test that show_correctness="never" hides correctness from learners and course staff.
+        """
+        self._xblock.show_correctness = ShowCorrectness.NEVER
+        assert not ProblemFeedbackService(self._xblock, user_is_staff=has_staff_access).correctness_available()
+
+    @ddt.data(
+        # Correctness not visible to learners if due date in the future
+        ('tomorrow', False, False),
+        # Correctness is visible to learners if due date in the past
+        ('yesterday', False, True),
+        # Correctness is visible to learners if due date in the past (just)
+        ('today', False, True),
+        # Correctness is visible to learners if there is no due date
+        (None, False, True),
+        # Correctness is visible to staff if due date in the future
+        ('tomorrow', True, True),
+        # Correctness is visible to staff if due date in the past
+        ('yesterday', True, True),
+        # Correctness is visible to staff if there is no due date
+        (None, True, True),
+    )
+    @ddt.unpack
+    def test_show_correctness_past_due(self, due_date_str, has_staff_access, expected_result):
+        """
+        Test show_correctness="past_due" to ensure:
+        * correctness is always visible to course staff
+        * correctness is always visible to everyone if there is no due date
+        * correctness is visible to learners after the due date, when there is a due date.
+        """
+        self._xblock.show_correctness = ShowCorrectness.PAST_DUE
+        if due_date_str is None:
+            self._xblock.due = None
+            self._xblock.close_date = None
+        else:
+            self._xblock.due = getattr(self, due_date_str)
+            self._xblock.close_date = getattr(self, due_date_str)
```

Comment:
The `closed_date` property should be provided by the `InheritanceMixin`. Isn't that the case here?

Reply:
In this case `self._xblock` is an instance of `MagicMock` class.

https://github.com/openedx/edx-platform/blob/5907b7e4f5ffa18dba4598871816b9fa353f8dcf/xmodule/tests/test_services.py#L172-L184